broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/AsyncSnapshotDirector.java

```diff
@@ -166,14 +173,16 @@ public static AsyncSnapshotDirector ofProcessingMode(
       final int partitionId,
       final StreamProcessor streamProcessor,
       final StateController stateController,
-      final Duration snapshotRate) {
+      final Duration snapshotRate,
+      final Supplier<CompletableFuture<Void>> flushLog) {
```

Comment:
:wrench: Supplier is a bit weird here, since were are not really interested in the result but more on just the action we execute. Maybe a Callable might make more sense.